homeassistant/components/tplink/sensor.py

```diff
@@ -69,43 +124,37 @@ class TPLinkSensorEntityDescription(SensorEntityDescription):
         device_class=SensorDeviceClass.ENERGY,
         state_class=SensorStateClass.TOTAL_INCREASING,
         precision=3,
+        value_fn=async_handle_emeter_attr,
     ),
     TPLinkSensorEntityDescription(
         key=ATTR_VOLTAGE,
         native_unit_of_measurement=UnitOfElectricPotential.VOLT,
         device_class=SensorDeviceClass.VOLTAGE,
         state_class=SensorStateClass.MEASUREMENT,
-        emeter_attr="voltage",
+        attribute_name="voltage",
+        value_fn=async_handle_emeter_attr,
         precision=1,
     ),
     TPLinkSensorEntityDescription(
         key=ATTR_CURRENT_A,
         native_unit_of_measurement=UnitOfElectricCurrent.AMPERE,
         device_class=SensorDeviceClass.CURRENT,
         state_class=SensorStateClass.MEASUREMENT,
-        emeter_attr="current",
+        attribute_name="current",
+        value_fn=async_handle_emeter_attr,
         precision=2,
     ),
+    TPLinkSensorEntityDescription(
+        key="on_since",
```

Comment:
I just realized that this information is only available if the device is on, so we may need to add a parameter to force the creation of sensors even if their value is `None` at the time of the initialization.

Also, I suppose this might also apply for the emeter sensors, too.